botocore/docs/bcdoc/style.py

```diff
@@ -229,9 +244,26 @@ def sphinx_reference_label(self, label, text=None):
         else:
             self.doc.write(text)
 
+    def _clean_link_text(self):
+        doc = self.doc
+        # Pop till we reach the link start character to retrieve link text.
+        last_write = doc.pop_write()[::-1]
+        while last_write[-1] != '`':
+            last_write += doc.pop_write()[
+                ::-1
+            ]  # Reverse text to preserve order
+        doc.push_write('`')
+        last_write = last_write[:-1]
+
+        # Remove whitespace and reverse again to correct order, then push clean text.
+        last_write = last_write.rstrip(' ')[::-1]
+        if last_write != '':
+            doc.push_write(last_write)
```

Comment:
I don't understand what all this reversing is for here. Isn't this just the same as the following?

```suggestion
        last_write = doc.pop_write()
        while not last_write.startswith('`'):
            last_write = doc.pop_write() + last_write
        doc.push_write('`')
        last_write = last_write[1:].rstrip(' ')
        if last_write != '':
            doc.push_write(last_write)
```

Reply:
Yup, this should achieve the same thing. I think I just ran with the first solution that clicked for me. Only think needed for this simplification is an `lstrip` instead of `rstrip` since we're trying to remove the space in an example such as 
```
-- Incorrect --
` Link Text <some_link>`

-- Correct --
`Link Text <some_link>`
```
I can also provide said example in a comment to help future readers.